resources/acsl/datastructures.md

```diff
@@ -48,21 +48,22 @@ different, as it is written as `X = POP()`. What this does is that it removes th
 element in the stack and stores it in variable X. So, if there was a stack with its
 top element being "E", then writing `H = POP()` would store "E" in H.
 
-In the case that the POP operation is called on an empty stack, then the variable is given the special value NIL, which means "nonexistent".
+In the case that the POP operation is called on an empty stack, then the variable is given the special value NIL(Not In List), which means "nonexistent".
 
 ## Queues
 
-Queues, on the other hand, are often used to handle requests. It's similar to waiting in line (or a *queue* in Britain); the first people in line are processed first. Hence, queues follow a "first in, first out" (FIFO) order.
+Queues, on the other hand, are often used to handle requests. It's similar to waiting in line (or a *queue*); the first people in line are processed first. Hence, queues follow a "first in, first out" (FIFO) order.
 
-The PUSH operation works the same as it would with stacks. However, with POP, instead of removing the top (most recently added) element, queues have their bottom element removed.
+The PUSH operation works the same as it would with stacks(the element is added to the top of the data structure). However, with POP, instead of removing the top (most recently added) element, queues have their bottom element removed.
 
+A tip to handle both queue and stack related problems is to keep a list of all the elements, with the pushed elements being added to the top of the list. In a queue, cross of the bottom element, and in a stack cross of the top element. 
```

Comment:
Cross off... nishikar stop using the wrong off lol

Reply:
It was one time lol